### ISSUE TITLE
Extra reportsUnnecessary suggestions for syntactically invalid file

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#25591]
From https://github.com/Microsoft/vscode/issues/52346

**TypeScript Version:**  3.0.0-dev.20180706

**Search Terms:** 
- reportsUnnecessary
- diagnostics 
- suggestions

**Code**
For the invalid code: 

```ts
export const _ = 1;

const a = 1 d;
```

**Bug**
See a suggestion diagnostic: 

```
  {
            "start": {
                "line": 3,
                "offset": 1
            },
            "end": {
                "line": 3,
                "offset": 15
            },
            "text": "All variables are unused.",
            "code": 6199,
            "category": "suggestion",
            "reportsUnnecessary": true
        }
```

This causes the entire line to become faded out. I do not believe we should report these errors when there are syntax errors

**Comment by mjbvz** [comment]
Another minimal example that has an extra unused diagnostic:

```ts
({
    a: 1
}
b: 1
})
```

`b` is reported as an unused label

**Comment by mhegazy** [comment]
Not sure what is the best solution here. @andy-ms thoughts?

**Comment by andy-ms** [comment]
It would probably be fine to never add suggestions when any parse errors exist.

**Comment by mjbvz** [comment]
Yes that sounds reasonable overall. Will have to see if it causes flickering in the editor though while users are writing
